zerver/lib/test_helpers.py

```diff
@@ -295,6 +295,13 @@ class HostRequestMock(HttpRequest):
     """A mock request object where get_host() works.  Useful for testing
     routes that use Zulip's subdomains feature"""
 
+    # The base class HttpRequest has GET and POST been immutable QueryDict, which makes
+    # it incompatible for HostRequestMock to define them as being mutable. Since the test
+    # cases rely on the mutability of these fields, we override it with type ignore.
```

Comment:
This is fine for now but I'd be curious how many tests change these vs. declaring a new HostRequestMock for new values; it seems possible we could either try to purge the modification pattern or make a `MutableHostRequestMock` used by a small number of tests that plan to do that.

Reply:
The main issue here is that `HostRequestMock` itself relies on this mutability of `GET` and `POST` to initialize them in the first place. `POST` is mutated all the type via `post_data`. For `GET`, it is not as often.